### ISSUE TITLE
Can't drop primary index. 

### ISSUE THREAD

**Issue by sergeycherepanov** [dolthub/dolt#7456]
When run next queries:

```sql
DROP TABLE IF EXISTS position_divisions;

CREATE TABLE `position_divisions` (
  `id` int unsigned NOT NULL AUTO_INCREMENT,
  `uuid` char(36) NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `uuid_unique` (`uuid`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;

DROP INDEX `PRIMARY` ON position_divisions;
``` 

The last cause an error:
```
[ERROR in query 4] `primary` does not exist as an index for this table
```

But the result of `SHOW INDEXES FROM position_divisions;`:
```
mysql> SHOW INDEXES FROM position_divisions;
+--------------------+------------+-------------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
| Table              | Non_unique | Key_name    | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment | Visible | Expression |
+--------------------+------------+-------------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
| position_divisions |          0 | PRIMARY     |            1 | id          | NULL      |           0 |     NULL | NULL   |      | BTREE      |         |               | YES     | NULL       |
| position_divisions |          0 | uuid_unique |            1 | uuid        | NULL      |           0 |     NULL | NULL   |      | BTREE      |         |               | YES     | NULL       |
+--------------------+------------+-------------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
2 rows in set (0.00 sec)
```

**Comment by sergeycherepanov** [comment]
Ok, I found the woking syntax for the query. But not I cant drop it because the field is AUTO_INCREMENT. Even in case I add an edditional index.

Tha target sql what I need
```sql
CREATE TABLE `position_divisions` (
  `id` int unsigned NOT NULL AUTO_INCREMENT,
  `uuid` char(36) NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `uuid_unique` (`uuid`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;

ALTER TABLE `position_divisions` ADD UNIQUE KEY `id` (`id`);
ALTER TABLE `position_divisions` DROP PRIMARY KEY;
ALTER TABLE `position_divisions` ADD PRIMARY KEY (`uuid`);
```
The error appear in attempt to drop primary index
```
[ERROR in query 4] error: incorrect table definition: there can be only one auto column and it must be defined as a key
Execution stopped!
```

**Comment by timsehn** [comment]
We'll get this syntax fixed. It's best to create a new table and select into it for now.

**Comment by sergeycherepanov** [comment]
hi @timsehn. I use the golang migration tool and applies already exists migrations to the doltdb. Yes, for now I fixed the migration. But re-creation of the table is not possible due to a lot of exists data and relations. I just created the issue to note about potential issue and for the workaroud if someone will facing with it in future.

**Comment by timsehn** [comment]
Great that you have a workaround. We'll get the syntax fixed Monday.

**Comment by timsehn** [comment]
I'm fairly certain this is related: https://github.com/dolthub/dolt/issues/2289

**Comment by fulghum** [comment]
Hi @sergeycherepanov, thank you for checking out Dolt! 

I did some quick testing with MySQL and I see that MySQL **does** support the `DROP INDEX` syntax with the `PRIMARY` index name, so we should get that gap fixed, too. 

I also see that as long as you add the `UNIQUE KEY` constraint to the table first, MySQL will allow you to drop the primary key. We will fix Dolt to match this behavior, too, which, as Tim pointed out, seems to be related to https://github.com/dolthub/dolt/issues/2289. 

Thank you for taking the tim to report this! I'll dig into these some more this afternoon and get started scoping the work out.

**Comment by fulghum** [comment]
Quick update on this one... I've got a fix at the `go-mysql-server` layer out for review, and am starting some work at the `dolt` layer. I'll keep you updated as we progress and get the fix released for you.

**Comment by fulghum** [comment]
@sergeycherepanov – the fix for being able to drop a primary key constraint on an `auto_increment` column has been released in [Dolt 1.33.0](https://github.com/dolthub/dolt/releases/tag/v1.33.0). Dolt now matches MySQL's behavior and allows you to drop the PK when another index is present that covers the `auto_increment` column. I ended up pulling out the support for dropping the PK by name using `DROP INDEX`, in order to get this fix out faster. If that syntax is important to you, let us know and we'll be happy to revisit that. 

Thank you for taking the time to report this issue. 🙏 Let us know if we can do anything else to help you build with Dolt!
